rfcs/text/0000-bundle-definition.md

```diff
@@ -0,0 +1,155 @@
+ 
+ [Hannes Schmidt](mailto:hannes@ucsc.edu)
+ 
+ [Andrey Kislyuk](mailto:akislyuk@chanzuckerberg.com)
+
+> Replace Mallory with Zina P and/or Chris V.
+
+## Shepherd
+***Leave this blank.** This role is assigned by DCP PM to guide the **Author(s)** through the RFC process.*
+
+*Recommended format for Shepherds:*
+
+ `[Name](mailto:username@example.com)`
+
+## Motivation
+
+*Describe the user or technical need in detail [with alignment to the DCP roadmap priorities where possible]. Link prior community discussions to demonstrate support for this RFC.*
+
+There are currently no documented definitions for what a "bundle" is in the HCA DCP, including no definitions of what they are, what they contain, or who they support. Lack of clarity causes confusion for data consumers (and DCP developers) about how the data and metadata in the DCP are structured and organized. At the June 2019 DCP F2F meeting in Cambridge, a breakout session was held to discuss "What is a bundle?", and a lot of useful discussion occurred. This RFC will formally document what a bundle is, anchoring the definition on how bundles support HCA DCP data consumers. This RFC will also describe new bundles types that could be supported by the HCA DCP.
+
+### User Stories
+
+*Share the [User Stories](https://www.mountaingoatsoftware.com/agile/user-stories) motivating this RFC.*
+
+1. As a data consumer who develops data processing tools (*e.g.* a sequence read aligner), I would like to browse, query, and get primary data from the DSS so that I can use it as testing data during development.
+1. As a data consumer who develops analysis tools (*e.g.* a clustering algorithm), I would like to browse, query, and get alignment data from the DSS so that I can use it as testing data during development.
+1. As a data consumer who is a computational biologist, I would like to get expression matrices from the DSS so that I can compare against data I already have.
+1. As a data consumer who is a biologist, I would like to get expression matrices from the DSS so that I can see the expression profile of &lt;my gene of interest&gt; in &lt;my cell type(s) or organ(s) of interest&gt;.
+1. As a data consumer who is a computational biologist, I would like to know what reference data was used to generate processed data in the DSS and I would like to be able to get it for myself so that I can process/analyze my own data with the same references as HCA DCP data.
+
+
+## Scientific "guardrails" [optional]
+
+*Describe recommended or mandated review from HCA Science governance to ensure that the RFC addresses the needs of the scientific community.*
+
+## Detailed Design
+
+*Explain the design in sufficient detail such that the implementation and (if appropriate) the interaction with existing DCP software are both reasonably clear.*
+
+### General bundle definition
+
+In the HCA DCP, a bundle is defined as: 
+1. a transaction of files that belong together for scientific reasons - **OR** - 
```

Comment:
As we adopt more and more scientific use cases this definition (1) will become less and less useful. With more use cases, files will need to be dynamically aggregated to fulfill specific use cases (as illustrated in the multlibrary RFA). If this is a hard grouping (a transaction of files) then would you group files only by run or only by release or biological network? None of these would be the right answer, but all would be needed groupings for scientific use cases.

Reply:
@TimothyTickle thanks for pointing this out. I updated and expanded the bundle type definition while merging the contents of #86 into this PR - please take another look.

I disagree that the "transactional unit" definition will become less useful. Operationally, a bundle is the output of an application run. It has to be grouped transactionally for data integrity puproses, and decorated with metadata for discoverability purposes. Whether the grouping is done by release, biological network, etc. is up to the application developers.

Reply:
agreed with @kislyuk ; transactions are key to data integrity and it is necessary for DCP components to think deeply about data integrity.